commands/src/main/java/org/wildfly/extras/creaper/commands/foundation/online/SnapshotBackup.java

```diff
@@ -35,20 +36,39 @@
         @Override
         public void apply(OnlineCommandContext occ) throws CommandFailedException, IOException, CliException {
             if (SnapshotBackup.this.snapshotName != null) {
-                throw new CommandFailedException("Snapshot was already backed up as "
+                throw new CommandFailedException("Snapshot was already taken: "
                         + SnapshotBackup.this.snapshotName);
             }
-            String snapshot = "";
+            String snapshotAbsolutePath = "";
             ModelNodeResult modelNodeResult = occ.client.execute(":take-snapshot");
             if (modelNodeResult.isSuccess() && modelNodeResult.hasDefinedValue()) {
-                snapshot = modelNodeResult.stringValue();
+                snapshotAbsolutePath = modelNodeResult.stringValue();
             }
             // returns absolute path, but we need just the name of snapshot
-            snapshot = snapshot.substring(snapshot.lastIndexOf(File.separator) + 1);
-            if (Strings.isNullOrEmpty(snapshot)) {
+            String snapshotName = getSnapshotName(snapshotAbsolutePath);
+            if (Strings.isNullOrEmpty(snapshotAbsolutePath)) {
                 throw new CommandFailedException("Unable to take snapshot");
             }
-            SnapshotBackup.this.snapshotName = snapshot;
+            SnapshotBackup.this.snapshotName = snapshotName;
+        }
+
+        /**
+         * Extract file name of absolute path to snapshot
+         *
+         * <p>
+         * Also handles situation when server and client are running on different platforms.
+         * </p>
+         *
+         * @param snapshotAbsolutePath Absolute path to snapshot
+         * @return name of snapshot
+         */
+        private String getSnapshotName(String snapshotAbsolutePath) {
```

Comment:
As the first line of the method body, I'd expect a comment along the lines of `Windows file names can't contain /, so if there's a / in the full path, it's not Windows`.